lib/models/BucketInfo.js

```diff
@@ -347,7 +347,7 @@ class BucketInfo {
      * does not have a bucket policy
      */
     getBucketPolicy() {
-        return this._bucketPolicy;
+        return JSON.parse(this._bucketPolicy);
```

Comment:
`this._bucketPolicy` might be undefined because `setBucketPolicy` doesn't have a check for valid values, and `JSON.parse` is unable to parse `undefined` 
```suggestion
        if (this._bucketPolicy === undefined) {
            return null;
        }
        return JSON.parse(this._bucketPolicy);
```

Reply:
Yeah, that's a good point. Following the format of some of our other classes like LifecycleConfiguration and ReplicationConfiguration, technically, this setter should never be called unless the validation of the, in this case policy, has already happened. It's just that that's done directly on the BucketPolicy class. It's definitely a confusing setup, but it looks like this:
`            const bucketPolicy = new BucketPolicy(request.post);
              const policyObj = bucketPolicy.getBucketPolicy();
              if (policyObj.error) {
                  return next(policyObj.error);
              }
              bucket.setBucketPolicy(bucketPolicy);
`
So the validation happens in the `bucketPolicy.getBucketPolicy()` step.
But you're 100% right, this should just be stored as an object, idk what I was thinking